File: apps/api/app/schemas/medicines.py
from datetime import date

from pydantic import Field

from app.schemas.base import CamelModel


class CreateMedicineRequest(CamelModel):
    name: str = Field(min_length=1, max_length=200)
    strength: str | None = Field(default=None, max_length=50)
    dosage: str | None = Field(default=None, max_length=100)
    frequency: str | None = Field(default=None, max_length=100)
    start_date: date | None = None
    end_date: date | None = None
    prescribing_doctor: str | None = Field(default=None, max_length=200)
    reason: str | None = Field(default=None, max_length=300)


class UpdateMedicineRequest(CamelModel):
    name: str | None = Field(default=None, min_length=1, max_length=200)
    strength: str | None = Field(default=None, max_length=50)
    dosage: str | None = Field(default=None, max_length=100)
    frequency: str | None = Field(default=None, max_length=100)
    start_date: date | None = None
    end_date: date | None = None
    prescribing_doctor: str | None = Field(default=None, max_length=200)
    reason: str | None = Field(default=None, max_length=300)
    active: bool | None = None


class MedicineResponse(CamelModel):
    id: str
    name: str
    strength: str | None
    dosage: str | None
    frequency: str | None
    start_date: date | None
    end_date: date | None
    prescribing_doctor: str | None
    reason: str | None
    active: bool

File: apps/web/lib/api/medicines.ts
import { apiFetch } from "./client";
import type { Medicine } from "./types";

export interface CreateMedicineInput {
  name: string;
  strength?: string;
  dosage?: string;
  frequency?: string;
  startDate?: string;
  endDate?: string;
  prescribingDoctor?: string;
  reason?: string;
}

export function createMedicine(input: CreateMedicineInput): Promise<Medicine> {
  return apiFetch<Medicine>("/medicines", { method: "POST", body: input });
}

export function listMedicines(): Promise<Medicine[]> {
  return apiFetch<Medicine[]>("/medicines");
}

export interface UpdateMedicineInput {
  active?: boolean;
  name?: string;
  strength?: string;
  dosage?: string;
  frequency?: string;
  endDate?: string;
  reason?: string;
}

export function updateMedicine(id: string, input: UpdateMedicineInput): Promise<Medicine> {
  return apiFetch<Medicine>(`/medicines/${id}`, { method: "PATCH", body: input });
}

export function deleteMedicine(id: string): Promise<void> {
  return apiFetch<void>(`/medicines/${id}`, { method: "DELETE" });
}
